src/modules/InstURL.rb

```diff
@@ -142,232 +83,45 @@ def RewriteCDUrl(url)
     # Convert install.inf to a URL useable by the package manager
     # @param [String] extra_dir append path to original URL
     # @return [String] new repository URL
-    def installInf2Url(extra_dir)
-      repo_url = ""
-
-      # bnc #406162
-      repo_url_from_inf = false
-      Builtins.foreach(["ZyppRepoURL"]) do |in_inf_key|
-        repo_url = Linuxrc.InstallInf(in_inf_key)
-        if repo_url != nil && repo_url != ""
-          Builtins.y2milestone(
-            "Using %1 directly from install.inf: %2",
-            in_inf_key,
-            URL.HidePassword(repo_url)
-          )
-          repo_url_from_inf = true
-          raise Break
-        end
-      end
+    def installInf2Url(extra_dir = "")
+      return @installInf2Url unless @installInf2Url.nil?
 
-      return repo_url if repo_url_from_inf
+      parts = URL.Parse(Linuxrc.InstallInf("ZyppRepoURL"))
 
-      # Initial repository hasn't been found in install.inf
-      # Trying to guess it ...
-      Builtins.y2warning("Initial repository not found in install.inf")
-
-      options = ""
-      url_tokens = {}
-
-      instmode = Linuxrc.InstallInf("InstMode") # mode
```

Comment:
`InstMode` is still documented in https://en.opensuse.org/SDB:Linuxrc_install.inf